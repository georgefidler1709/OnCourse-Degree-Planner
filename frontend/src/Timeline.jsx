import React, {Component} from 'react';
import '@atlaskit/css-reset';
import styled from 'styled-components';
import { DragDropContext } from 'react-beautiful-dnd';
import mockDegree from './mockDegree';
import Term from './Term';

const Container = styled.div`
  display: flex;
`;

class Timeline extends Component {
  constructor(props) {
    //console.log(props.location.state.plan)
    super(props)
    this.state = mockDegree //props.location.state.plan
  }

  onDragEnd = result => {
    // TODO: preserve reorder of terms
  };

  render() {
    return (
      <DragDropContext onDragEnd={this.onDragEnd}>
        {this.state.yearOrder.map(yearId => {
          const year = this.state.years[yearId];

          return (
            <Container>
              {year.termOrder.map(termId => {
                const term = this.state.terms[termId];
                const courses = term.courseIds.map(courseId => this.state.courses[courseId]);
      
                return <Term key={term.id} term={term} courses={courses} />;
              })}
            </Container>
          )
        })}
      </DragDropContext>
    );
  }
}

export default Timeline
