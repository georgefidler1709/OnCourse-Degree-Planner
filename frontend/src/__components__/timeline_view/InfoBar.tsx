import React from 'react';
import styled from 'styled-components';
import CourseDropBox from "./CourseDropBox"

const Container = styled.div`
  margin: 8px;
  padding: 8px;
  border: 1px solid lightgrey;
  border-radius: 2px;
  width: 400px;
  display: flex;
  flex-direction: column;
  text-align: center;
  position: fixed;
`;

const Title = styled.h3`
  padding: 8px;
`;

const SubTitle = styled.h5`
  padding: 4px;
`;

interface Req {
  filter_type: string;
  units: number;
  info: string;
}

interface InfoBarProps {
  degree_id: number;
  degree_name: string;
  degree_reqs: Array<Req>
}

function InfoBar(props: InfoBarProps) {
  return (
    <Container>
      <header>
        <Title>{props.degree_name}</Title>
        <Title>{props.degree_id.toString()}</Title>
      </header>
      <body>
        <SubTitle>Requirements</SubTitle>
        {props.degree_reqs.map(req => { return (
          <div>
            <p>{`${req.filter_type}: ${req.units} UOC of`}</p>
            <ul>
              <li>{`${req.info}`}</li>
            </ul>
          </div>
        )
        })}
      </body>
      <footer>
        <CourseDropBox type="Add"/>
      </footer>
    </Container>
  );
}

export default InfoBar
