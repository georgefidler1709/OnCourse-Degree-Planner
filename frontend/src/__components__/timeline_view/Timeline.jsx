import React, {Component} from 'react';
import '@atlaskit/css-reset';
import styled from 'styled-components';
import { DragDropContext } from 'react-beautiful-dnd';
import Term from './Term';

const Container = styled.div`
  display: flex;
`;

class Timeline extends Component {

  constructor(props) {
    super(props)
    this.state = props.location.state.plan
  }

  onDragEnd = result => {
    // TODO: preserve reorder of terms
  };

  getCourseInfo(course_id) {
    const matching_course = 
      this.state.courses.find((course) => course.subject + course.code.toString() === course_id)
    
    return {course_id: course_id, course_info: matching_course}
  }

  render() {
    const program = this.state.program

    // fill in required years for the program duration
    let timeline = []
    let year = program.enrollments[0].year
    for(let i = 0; i < program.duration; ++i) {
      timeline.push(year++)
    }

    return (
      <DragDropContext onDragEnd={this.onDragEnd}>
        {timeline.map((year_num, year_index) => {
          let year = {term_plans: [], year: year_num}
          if(year_index < program.enrollments.length) {
            year = program.enrollments[year_index]
          } 

          // fill in the minimum 3 terms per year
          const required_terms = 3
          let cur_term = 1
          let terms = year.term_plans.map(term => {
            if(term.term === cur_term) ++cur_term
            return term
          })
          for( ; cur_term <= required_terms; ++cur_term) terms.push({course_ids: [], term: cur_term})

          return (
            <Container>
              {terms.map(term => {
                const courses = term.course_ids.map(course_id => this.getCourseInfo(course_id));
                const term_tag = "T" + term.term.toString() + " " + year_num.toString()
                return <Term key={term_tag} term_id={term_tag} courses={courses} />;
              })}
            </Container>
          )
        })}
      </DragDropContext>
    );
  }
}

export default Timeline
