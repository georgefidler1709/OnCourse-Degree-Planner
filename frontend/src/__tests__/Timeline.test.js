import React from 'react'
import { shallow, mount } from 'enzyme';
import { Button } from 'react-bootstrap'
import Timeline from '../__components__/timeline_view/Timeline';
import InfoBar from '../__components__/timeline_view/InfoBar';

console.error = jest.fn();
console.warn = jest.fn();
console.log = jest.fn();

function sleep(ms) {
  return new Promise(resolve => setTimeout(resolve, ms));
}

describe('savePlan method', () => {
  it('saves a plan when save button is pressed', async() => {
    const wrapper = mount(<Timeline match={{params: {degree: "degree"}}} />);
    await sleep(100);
    wrapper.update();

    const instance = wrapper.instance();
    instance.savePlan = jest.fn()
    const spy = jest.spyOn(instance, 'savePlan');
    wrapper.instance().forceUpdate();
    wrapper.find(Button).first().simulate('click');
    expect(spy).toHaveBeenCalled();

    wrapper.unmount();
  });
});

describe('onDragEnd method', () => {
  it('will preserve changing of the order of courses within a term', async() => {
    const wrapper = shallow(<Timeline match={{params: {degree: "degree"}}} />);
    await sleep(1000);
    wrapper.update();

    let source_course = wrapper.state().program.enrollments[0].term_plans[0].course_ids[0]
    let source_length = wrapper.state().program.enrollments[0].term_plans[0].course_ids.length
    let new_source_course = wrapper.state().program.enrollments[0].term_plans[0].course_ids[1]

    wrapper.instance().onDragEnd({
      destination: {index: 2, droppableId: "1 2019"},
      source: {index: 0, droppableId: "1 2019"},
      draggableId: source_course,
    })
    expect(wrapper.state().program.enrollments[0].term_plans[0].course_ids[0]).toBe(new_source_course)
    expect(wrapper.state().program.enrollments[0].term_plans[0].course_ids[2]).toBe(source_course)
    expect(wrapper.state().program.enrollments[0].term_plans[0].course_ids.length).toBe(source_length)
  });
  it('will preserve changing the term of a course', async() => {
    const wrapper = shallow(<Timeline match={{params: {degree: "degree"}}} />);
    await sleep(1000);
    wrapper.update();
    let source_course = wrapper.state().program.enrollments[0].term_plans[0].course_ids[0]
    let new_source_course = wrapper.state().program.enrollments[0].term_plans[0].course_ids[1]
    let source_length = wrapper.state().program.enrollments[0].term_plans[0].course_ids.length
    let dest_length = wrapper.state().program.enrollments[0].term_plans[1].course_ids.length

    wrapper.instance().onDragEnd({
      destination: {index: 0, droppableId: "2 2019"},
      source: {index: 0, droppableId: "1 2019"},
      draggableId: source_course,
    })
    expect(wrapper.state().program.enrollments[0].term_plans[0].course_ids[0]).toBe(new_source_course)
    expect(wrapper.state().program.enrollments[0].term_plans[0].course_ids).toHaveLength(source_length - 1)
    expect(wrapper.state().program.enrollments[0].term_plans[1].course_ids[0]).toBe(source_course)
    expect(wrapper.state().program.enrollments[0].term_plans[1].course_ids).toHaveLength(dest_length + 1)

    // reverse operation
    wrapper.instance().onDragEnd({
      destination: {index: 0, droppableId: "1 2019"},
      source: {index: 0, droppableId: "2 2019"},
      draggableId: "COMP1511",
    })
  });
});

describe('addMissingTerms method', () => {
  it('will add missing term', async() => {
    const wrapper = shallow(<Timeline match={{params: {degree: "degree"}}} />);
    await sleep(1000);
    wrapper.update();
    wrapper.state().program.enrollments[0].term_plans.pop()
    expect(wrapper.state().program.enrollments[0].term_plans).toHaveLength(2)
    wrapper.instance().addMissingTerms()
    expect(wrapper.state().program.enrollments[0].term_plans).toHaveLength(3)
  });

  it('will add a missing year', async() => {
    const wrapper = shallow(<Timeline match={{params: {degree: "degree"}}} />);
    await sleep(1000);
    wrapper.update();
    wrapper.state().program.enrollments.pop()
    expect(wrapper.state().program.enrollments).toHaveLength(2)
    wrapper.instance().addMissingTerms()
    expect(wrapper.state().program.enrollments).toHaveLength(3)
  });
});

describe('onDragStart method', () => {
  it('will highlight terms which contain an offering for the course being dragged', async() => {
    const wrapper = shallow(<Timeline match={{params: {degree: "degree"}}} />);
    await sleep(100);
    wrapper.update();

    wrapper.instance().onDragStart({draggableId: "COMP3121", source: {droppableId: "1 2019"}})
    expect(wrapper.state().program.enrollments[0].term_plans[0].highlight).toBeFalsy()
    await sleep(100);
    wrapper.update();
    expect(wrapper.state().program.enrollments[0].term_plans[1].highlight).toBeTruthy()
  });
});

describe('Render degree planning timeline view', () => {
  it('renders correctly', async() => {
    const wrapper = shallow(<Timeline match={{params: {degree: "degree"}}} />);
    await sleep(100);
    wrapper.update();

    expect(wrapper).toMatchSnapshot(); 

    wrapper.unmount();
  })
});

describe('removeCourse method', () => {
  it('has the course to be removed before removal', async() => {
    const wrapper = mount(<Timeline match={{params: {degree: "degree"}}} />);
    await sleep(100);
    wrapper.update();
    expect(wrapper.state().courses["COMP1511"]).toBeDefined()
  });

  it('does not have removed course after removal', async() => {
    const wrapper = mount(<Timeline match={{params: {degree: "degree"}}} />);
    await sleep(100);
    wrapper.update();
    wrapper.instance().removeCourse("COMP1511")
    expect(wrapper.state().courses["COMP1511"]).toBeUndefined()
  });
});