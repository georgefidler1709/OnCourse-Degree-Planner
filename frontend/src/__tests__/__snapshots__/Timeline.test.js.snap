// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render degree planning timeline view renders correctly 1`] = `
<div>
  <Navbar
    bg="dark"
    collapseOnSelect={false}
    expand={true}
    id="navbar"
    variant="dark"
  >
    <NavbarBrand
      href="/"
    >
      OnCourse
    </NavbarBrand>
    <Nav
      className="mr-auto"
      fill={false}
      justify={false}
    />
    <Styled(Button)
      id="save"
      onClick={[Function]}
      variant="outline-info"
    >
      <i
        className="fa fa-save"
      />
    </Styled(Button)>
    <Styled(Button)
      variant="outline-info"
    >
      <i
        className="fa fa-cog"
      />
    </Styled(Button)>
  </Navbar>
  <br />
  <styled.div>
    <DragDropContext
      onDragEnd={[Function]}
      onDragStart={[Function]}
    >
      <div>
        <styled.div
          id="timeline"
        >
           
          <div>
            <styled.div
              key="2019"
            >
              <Term
                courses={
                  Array [
                    Object {
                      "code": "MATH1081",
                      "coreqs": "(DPST1013 OR MATH1131 OR MATH1141 OR MATH1151)",
                      "equivalents": "",
                      "exclusions": "MATH1090",
                      "name": "Discrete Mathematics",
                      "prereqs": "",
                      "terms": Array [
                        Object {
                          "term": 1,
                          "year": 2019,
                        },
                        Object {
                          "term": 2,
                          "year": 2019,
                        },
                        Object {
                          "term": 3,
                          "year": 2019,
                        },
                        Object {
                          "term": 1,
                          "year": 2020,
                        },
                        Object {
                          "term": 2,
                          "year": 2020,
                        },
                        Object {
                          "term": 3,
                          "year": 2020,
                        },
                        Object {
                          "term": 1,
                          "year": 2021,
                        },
                        Object {
                          "term": 2,
                          "year": 2021,
                        },
                        Object {
                          "term": 3,
                          "year": 2021,
                        },
                        Object {
                          "term": 1,
                          "year": 2022,
                        },
                        Object {
                          "term": 2,
                          "year": 2022,
                        },
                        Object {
                          "term": 3,
                          "year": 2022,
                        },
                        Object {
                          "term": 1,
                          "year": 2023,
                        },
                        Object {
                          "term": 2,
                          "year": 2023,
                        },
                        Object {
                          "term": 3,
                          "year": 2023,
                        },
                        Object {
                          "term": 1,
                          "year": 2024,
                        },
                        Object {
                          "term": 2,
                          "year": 2024,
                        },
                        Object {
                          "term": 3,
                          "year": 2024,
                        },
                        Object {
                          "term": 1,
                          "year": 2025,
                        },
                        Object {
                          "term": 2,
                          "year": 2025,
                        },
                        Object {
                          "term": 3,
                          "year": 2025,
                        },
                      ],
                      "units": 6,
                    },
                    Object {
                      "code": "COMP1511",
                      "coreqs": "",
                      "equivalents": "",
                      "exclusions": "DPST1091",
                      "name": "Programming Fundamentals",
                      "prereqs": "",
                      "terms": Array [
                        Object {
                          "term": 1,
                          "year": 2019,
                        },
                        Object {
                          "term": 2,
                          "year": 2019,
                        },
                        Object {
                          "term": 3,
                          "year": 2019,
                        },
                        Object {
                          "term": 1,
                          "year": 2020,
                        },
                        Object {
                          "term": 2,
                          "year": 2020,
                        },
                        Object {
                          "term": 3,
                          "year": 2020,
                        },
                        Object {
                          "term": 1,
                          "year": 2021,
                        },
                        Object {
                          "term": 2,
                          "year": 2021,
                        },
                        Object {
                          "term": 3,
                          "year": 2021,
                        },
                        Object {
                          "term": 1,
                          "year": 2022,
                        },
                        Object {
                          "term": 2,
                          "year": 2022,
                        },
                        Object {
                          "term": 3,
                          "year": 2022,
                        },
                        Object {
                          "term": 1,
                          "year": 2023,
                        },
                        Object {
                          "term": 2,
                          "year": 2023,
                        },
                        Object {
                          "term": 3,
                          "year": 2023,
                        },
                        Object {
                          "term": 1,
                          "year": 2024,
                        },
                        Object {
                          "term": 2,
                          "year": 2024,
                        },
                        Object {
                          "term": 3,
                          "year": 2024,
                        },
                        Object {
                          "term": 1,
                          "year": 2025,
                        },
                        Object {
                          "term": 2,
                          "year": 2025,
                        },
                        Object {
                          "term": 3,
                          "year": 2025,
                        },
                      ],
                      "units": 6,
                    },
                  ]
                }
                key="1 2019"
                removeCourse={[Function]}
                termId="1 2019"
              />
              <Term
                courses={
                  Array [
                    Object {
                      "code": "MATH1131",
                      "coreqs": "",
                      "equivalents": "DPST1013",
                      "exclusions": "(DPST1013 AND ECON1202 AND ECON2291 AND MATH1011 AND MATH1031 AND MATH1141 AND MATH1151)",
                      "name": "Mathematics 1A",
                      "prereqs": "",
                      "terms": Array [
                        Object {
                          "term": 1,
                          "year": 2019,
                        },
                        Object {
                          "term": 2,
                          "year": 2019,
                        },
                        Object {
                          "term": 3,
                          "year": 2019,
                        },
                        Object {
                          "term": 1,
                          "year": 2020,
                        },
                        Object {
                          "term": 2,
                          "year": 2020,
                        },
                        Object {
                          "term": 3,
                          "year": 2020,
                        },
                        Object {
                          "term": 1,
                          "year": 2021,
                        },
                        Object {
                          "term": 2,
                          "year": 2021,
                        },
                        Object {
                          "term": 3,
                          "year": 2021,
                        },
                        Object {
                          "term": 1,
                          "year": 2022,
                        },
                        Object {
                          "term": 2,
                          "year": 2022,
                        },
                        Object {
                          "term": 3,
                          "year": 2022,
                        },
                        Object {
                          "term": 1,
                          "year": 2023,
                        },
                        Object {
                          "term": 2,
                          "year": 2023,
                        },
                        Object {
                          "term": 3,
                          "year": 2023,
                        },
                        Object {
                          "term": 1,
                          "year": 2024,
                        },
                        Object {
                          "term": 2,
                          "year": 2024,
                        },
                        Object {
                          "term": 3,
                          "year": 2024,
                        },
                        Object {
                          "term": 1,
                          "year": 2025,
                        },
                        Object {
                          "term": 2,
                          "year": 2025,
                        },
                        Object {
                          "term": 3,
                          "year": 2025,
                        },
                      ],
                      "units": 6,
                    },
                    Object {
                      "code": "COMP1521",
                      "coreqs": "",
                      "equivalents": "",
                      "exclusions": "DPST1092",
                      "name": "Computer Systems Fundamentals",
                      "prereqs": "(COMP1911 OR COMP1917 OR COMP1511 OR DPST1091)",
                      "terms": Array [
                        Object {
                          "term": 2,
                          "year": 2019,
                        },
                        Object {
                          "term": 3,
                          "year": 2019,
                        },
                        Object {
                          "term": 2,
                          "year": 2020,
                        },
                        Object {
                          "term": 3,
                          "year": 2020,
                        },
                        Object {
                          "term": 2,
                          "year": 2021,
                        },
                        Object {
                          "term": 3,
                          "year": 2021,
                        },
                        Object {
                          "term": 2,
                          "year": 2022,
                        },
                        Object {
                          "term": 3,
                          "year": 2022,
                        },
                        Object {
                          "term": 2,
                          "year": 2023,
                        },
                        Object {
                          "term": 3,
                          "year": 2023,
                        },
                        Object {
                          "term": 2,
                          "year": 2024,
                        },
                        Object {
                          "term": 3,
                          "year": 2024,
                        },
                        Object {
                          "term": 2,
                          "year": 2025,
                        },
                        Object {
                          "term": 3,
                          "year": 2025,
                        },
                      ],
                      "units": 6,
                    },
                    Object {
                      "code": "MATH1231",
                      "coreqs": "",
                      "equivalents": "DPST1014",
                      "exclusions": "(DPST1014 AND MATH1021 AND MATH1241 AND MATH1251)",
                      "name": "Mathematics 1B",
                      "prereqs": "(DPST1013 OR MATH1131 OR MATH1141)",
                      "terms": Array [
                        Object {
                          "term": 1,
                          "year": 2019,
                        },
                        Object {
                          "term": 2,
                          "year": 2019,
                        },
                        Object {
                          "term": 3,
                          "year": 2019,
                        },
                        Object {
                          "term": 1,
                          "year": 2020,
                        },
                        Object {
                          "term": 2,
                          "year": 2020,
                        },
                        Object {
                          "term": 3,
                          "year": 2020,
                        },
                        Object {
                          "term": 1,
                          "year": 2021,
                        },
                        Object {
                          "term": 2,
                          "year": 2021,
                        },
                        Object {
                          "term": 3,
                          "year": 2021,
                        },
                        Object {
                          "term": 1,
                          "year": 2022,
                        },
                        Object {
                          "term": 2,
                          "year": 2022,
                        },
                        Object {
                          "term": 3,
                          "year": 2022,
                        },
                        Object {
                          "term": 1,
                          "year": 2023,
                        },
                        Object {
                          "term": 2,
                          "year": 2023,
                        },
                        Object {
                          "term": 3,
                          "year": 2023,
                        },
                        Object {
                          "term": 1,
                          "year": 2024,
                        },
                        Object {
                          "term": 2,
                          "year": 2024,
                        },
                        Object {
                          "term": 3,
                          "year": 2024,
                        },
                        Object {
                          "term": 1,
                          "year": 2025,
                        },
                        Object {
                          "term": 2,
                          "year": 2025,
                        },
                        Object {
                          "term": 3,
                          "year": 2025,
                        },
                      ],
                      "units": 6,
                    },
                  ]
                }
                key="2 2019"
                removeCourse={[Function]}
                termId="2 2019"
              />
              <Term
                courses={Array []}
                highlight={false}
                key="3 2019"
                removeCourse={[Function]}
                termId="3 2019"
              />
            </styled.div>
          </div>
          <div>
            <styled.div
              key="2020"
            >
              <Term
                courses={Array []}
                highlight={false}
                key="1 2020"
                removeCourse={[Function]}
                termId="1 2020"
              />
              <Term
                courses={
                  Array [
                    Object {
                      "code": "COMP3121",
                      "coreqs": "",
                      "equivalents": "COMP3821
COMP9801
COMP3120
COMP9101",
                      "exclusions": "",
                      "name": "Algorithms and Programming Techniques",
                      "prereqs": "(COMP1927 OR COMP2521)",
                      "terms": Array [
                        Object {
                          "term": 2,
                          "year": 2019,
                        },
                        Object {
                          "term": 2,
                          "year": 2020,
                        },
                        Object {
                          "term": 2,
                          "year": 2021,
                        },
                        Object {
                          "term": 2,
                          "year": 2022,
                        },
                        Object {
                          "term": 2,
                          "year": 2023,
                        },
                        Object {
                          "term": 2,
                          "year": 2024,
                        },
                        Object {
                          "term": 2,
                          "year": 2025,
                        },
                      ],
                      "units": 6,
                    },
                    Object {
                      "code": "COMP2511",
                      "coreqs": "",
                      "equivalents": "",
                      "exclusions": "(COMP2011 AND COMP2911)",
                      "name": "Object-Oriented Design & Programming",
                      "prereqs": "COMP1531
(COMP1927 OR COMP2521)",
                      "terms": Array [
                        Object {
                          "term": 2,
                          "year": 2019,
                        },
                        Object {
                          "term": 3,
                          "year": 2019,
                        },
                        Object {
                          "term": 2,
                          "year": 2020,
                        },
                        Object {
                          "term": 3,
                          "year": 2020,
                        },
                        Object {
                          "term": 2,
                          "year": 2021,
                        },
                        Object {
                          "term": 3,
                          "year": 2021,
                        },
                        Object {
                          "term": 2,
                          "year": 2022,
                        },
                        Object {
                          "term": 3,
                          "year": 2022,
                        },
                        Object {
                          "term": 2,
                          "year": 2023,
                        },
                        Object {
                          "term": 3,
                          "year": 2023,
                        },
                        Object {
                          "term": 2,
                          "year": 2024,
                        },
                        Object {
                          "term": 3,
                          "year": 2024,
                        },
                        Object {
                          "term": 2,
                          "year": 2025,
                        },
                        Object {
                          "term": 3,
                          "year": 2025,
                        },
                      ],
                      "units": 6,
                    },
                  ]
                }
                key="2 2020"
                removeCourse={[Function]}
                termId="2 2020"
              />
              <Term
                courses={Array []}
                highlight={false}
                key="3 2020"
                removeCourse={[Function]}
                termId="3 2020"
              />
            </styled.div>
          </div>
          <div>
            <styled.div
              key="2021"
            >
              <Term
                courses={Array []}
                highlight={false}
                key="1 2021"
                removeCourse={[Function]}
                termId="1 2021"
              />
              <Term
                courses={Array []}
                highlight={false}
                key="2 2021"
                removeCourse={[Function]}
                termId="2 2021"
              />
              <Term
                courses={Array []}
                highlight={false}
                key="3 2021"
                removeCourse={[Function]}
                termId="3 2021"
              />
            </styled.div>
          </div>
           
        </styled.div>
        <styled.div>
          <InfoBar
            add_event={[Function]}
            degree_id="3778"
            degree_name="Computer Science"
            degree_reqs={
              Array [
                Object {
                  "filter_type": "GenEd",
                  "units": 12,
                },
                Object {
                  "filter_type": "FreeElective",
                  "units": 36,
                },
              ]
            }
            remove_course={[Function]}
          />
        </styled.div>
      </div>
    </DragDropContext>
  </styled.div>
</div>
`;
